src/OpenTK/Platform/Egl/EglContext.cs

```diff
@@ -82,36 +82,13 @@ internal abstract class EglContext : EmbeddedGraphicsContext
             {
                 Debug.Print("[EGL] Failed to bind rendering API. Error: {0}", Egl.GetError());
             }
-
-            bool offscreen = (flags & GraphicsContextFlags.Offscreen) != 0;
-
-            SurfaceType surfaceType = offscreen
-                ? SurfaceType.PBUFFER_BIT
-                : SurfaceType.WINDOW_BIT;
-
-            Mode = new EglGraphicsMode().SelectGraphicsMode(surfaceType,
-                    window.Display, mode.ColorFormat, mode.Depth, mode.Stencil,
-                    mode.Samples, mode.AccumulatorFormat, mode.Buffers, mode.Stereo,
-                    Renderable);
-
+            Mode = SelectGraphicsMode(window, mode, flags, Renderable);
             if (!Mode.Index.HasValue)
             {
                 throw new GraphicsModeException("Invalid or unsupported GraphicsMode.");
             }
             IntPtr config = Mode.Index.Value;
-
-            if (window.Surface == IntPtr.Zero)
-            {
-                if (!offscreen)
-                {
-                    window.CreateWindowSurface(config);
-                }
-                else
-                {
-                    window.CreatePbufferSurface(config);
-                }
-            }
-
+            EnsureSurface();
```

Comment:
Here we are not using the `Mode` that we got just a few lines earlier. I would make `EnsureSurface` take the mode as an argument.